src/net.cpp

```diff
@@ -2880,3 +2880,23 @@ uint64_t CConnman::CalculateKeyedNetGroup(const CAddress& ad) const
 
     return GetDeterministicRandomizer(RANDOMIZER_ID_NETGROUP).Write(vchNetGroup.data(), vchNetGroup.size()).Finalize();
 }
+
+void LogMessage(const CNode& node, const std::string& msg_type, const Span<const unsigned char>& data, bool is_incoming)
+{
+    auto time = GetTime<std::chrono::microseconds>();
+
+    fs::path base_path = GetDataDir() / "message_logging" / node.addr.ToString();
+    fs::create_directories(base_path);
+
+    fs::path path = base_path / (is_incoming ? "msgs_recv.dat" : "msgs_sent.dat");
```

Comment:
I don't think ".dat" is the right extension. These are logs, not data, and should end in ".log".

Reply:
I think `.dat` is fine since this is a non-text binary file. Personally I expect files with the `.log` suffix to be enjoyable in `less` (more or less) :)

Reply:
Yes, please disregard my comment, I misunderstood!